WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -86,7 +87,38 @@ final class TotalsViewModel: ObservableObject {
         startSyncingOrder(with: cartItems, allItems: allItems)
     }
 
+    private func areOrderAndCartDifferent(cartItems: [CartItem]) -> Bool {
+        // check if order has same items as cart does
+        if let order {
+            // first we get list of all products 1 by 1
+            var cleanOrderItems: [POSOrderItem] = order.items.flatMap {
+                Array(repeating: $0, count: $0.quantity.intValue)
+            }
+            var cleanCartItems: [POSItem] = cartItems.flatMap {
+                Array(repeating: $0.item, count: $0.quantity)
+            }
```

Comment:
My understanding of this bit is that we're re-mapping all objects into an ordered list based on their quantities, right? So an `item A with a quantity of 2`, and an `item B with a quantity of 3` will output something like `[Item A, Item A, Item B, Item B, Item B]`, correct?

In this case maybe we should rename the function to something that reflects the purpose of the function, rather than `clean`. 

Since we're repeating logic here for the interfaces, maybe we could use a generic approach if we can, or rely the methods to the own protocols/implementations, rather than having the logic here, which would also clear up the method quite a bit. Eg:

```
func doSomeItemReorg<T>(_ items: [T], quantity: (T) -> Int) -> [T] {
	return items.flatMap { item in
		Array(repeating: item, count: quantity(item))
	}
}
```

As we need them sorted later on for the rest of the logic, the sorting logic could also be added as a helper method to the object.

Overall I agree with @soundsokay that this should be split in different smaller bits and be unit tested, otherwise we won't know what happens in the function internals and what's the expected input/output.